src/Core/Managed/Net40/Extensibility/Implementation/TelemetryConfigurationFactory.cs

```diff
@@ -44,8 +44,27 @@ protected TelemetryConfigurationFactory()
         /// </remarks>
         public static TelemetryConfigurationFactory Instance
         {
-            get { return instance ?? (instance = new TelemetryConfigurationFactory()); }
-            set { instance = value; }
+            // Both get and set should write something to CoreEventSource so as to initialize it. This is a fix for below deadlock issue   
```

Comment:
1. This is kind of a hack that isn't guaranteed to be present or work in future as code is refactored which is worrisome.
2. Code comments should not refer to bugs or issues through links, instead describe the situation in the comment so it doesn't rely on an external link that might not stay valid.